lwip-tests/build.sh

```diff
@@ -0,0 +1,94 @@
+
+# Download check to lwip-tests directory (only lwip-tests use it)
+PREFIX_LWIP_TEST="${PREFIX_PROJECT}/phoenix-rtos-ports/lwip-tests"
+PREFIX_LWIP_TEST_BUILD="${PREFIX_BUILD}/lwip-test"
+PREFIX_CHECK_BUILD="${PREFIX_LWIP_TEST_BUILD}/check"
+PREFIX_CHECK_SRC="${PREFIX_CHECK_BUILD}/$CHECK"
+
+#
+# Download and unpack
+#
+if [ ! -f "${PREFIX_LWIP_TEST}/${CHECK}.tar.gz" ]; then
+	if ! wget "$CHECK_PKG_URL" -O "${PREFIX_LWIP_TEST}/${CHECK}.tar.gz" --no-check-certificate; then
+		wget "$CHECK_PKG_MIRROR_URL" -P "$PREFIX_LWIP_TEST" --no-check-certificate
+	fi
+fi
+
+mkdir -p "$PREFIX_CHECK_BUILD"
+if [ ! -d "$PREFIX_CHECK_SRC" ]; then
+	tar xzf "${PREFIX_LWIP_TEST}/${CHECK}.tar.gz" -C "$PREFIX_CHECK_BUILD" && \
+	mv "${PREFIX_CHECK_BUILD}/check-$CHECK_COMMIT" "${PREFIX_CHECK_SRC}"
+fi
+
+#
+# Configure
+#
+mkdir -p "${PREFIX_CHECK_BUILD}/include"
+if [ ! -f "{$PREFIX_CHECK_BUILD}/config.h" ]; then
+	CHECK_CFLAGS="-std=gnu99 -I${PREFIX_CHECK_BUILD}/include"
+	(cd "$PREFIX_CHECK_SRC" && autoreconf -i -v -f && \
+	PKG_CONFIG="" ./configure CC="$CC" LD="$LD" AR="$AR" STRIP="${CROSS}strip" CFLAGS="$CFLAGS $CHECK_CFLAGS" \
+							  LDFLAGS="$LDFLAGS" --host="${HOST}" --build="x86_64-linux-gnu" --disable-subunit \
+							  --disable-shared --disable-doc --enable-static --libdir="$PREFIX_A" \
+							  --includedir="${PREFIX_CHECK_BUILD}/include")
+fi
+
+#
+# Make
+#
+SUBDIRS="lib src . tests" # Run makefiles from this directories
+make -C "$PREFIX_CHECK_SRC"
+make SUBDIRS="$SUBDIRS" doc_DATA="" m4data_DATA="" -C "$PREFIX_CHECK_SRC" install
+
+b_log "Building lwip-test"
+
+LWIP_CONTRIB_COMMIT="35b011d4cf4c4b480f8859c456587a884ec9d287"
+LWIP_CONTRIB_PKG_URL="https://github.com/particle-iot/lwip-contrib/archive/${LWIP_CONTRIB_COMMIT}.tar.gz"
+LWIP_CONTRIB_PKG_MIRROR_URL="https://files.phoesys.com/ports/lwip-contrib.tar.gz"
+PREFIX_LWIP_CONTRIB="${PREFIX_LWIP_TEST_BUILD}/contrib-lwip"
+PREFIX_LWIP_CHECK="${PREFIX_LWIP_CONTRIB}/ports/unix/check"
+PREFIX_LWIP_TEST_MARKERS="${PREFIX_LWIP_TEST_BUILD}/markers"
+
+#
+# Download and unpack lwip-contrib
+#
+if [ ! -f "${PREFIX_LWIP_TEST}/lwip-contrib.tar.gz" ]; then
+	if ! wget "$LWIP_CONTRIB_PKG_URL" -O "${PREFIX_LWIP_TEST}/lwip-contrib.tar.gz" --no-check-certificate; then
+		wget "$LWIP_CONTRIB_PKG_MIRROR_URL" -P "$PREFIX_LWIP_TEST" --no-check-certificate
+	fi
+fi
+
+if [ ! -d "$PREFIX_LWIP_CONTRIB" ]; then
+	tar xzf "${PREFIX_LWIP_TEST}/lwip-contrib.tar.gz" -C "$PREFIX_LWIP_TEST_BUILD" && \
+	mv "${PREFIX_LWIP_TEST_BUILD}/lwip-contrib-$LWIP_CONTRIB_COMMIT" "${PREFIX_LWIP_TEST_BUILD}/contrib-lwip"
+fi
+
+#
+# Apply patches
+#
+mkdir -p "$PREFIX_LWIP_TEST_MARKERS"
+for patchfile in "$PREFIX_LWIP_TEST"/patch/*; do
+	if [ ! -f "${PREFIX_LWIP_TEST_MARKERS}/$(basename "$patchfile").applied" ]; then
+		echo "applying patch: $patchfile"
+		patch -d "${PREFIX_LWIP_CONTRIB}/" -p1 < "$patchfile"
+		touch "${PREFIX_LWIP_TEST_MARKERS}/$(basename "$patchfile").applied"
+	fi
+done
+
+make clean check -C "$PREFIX_LWIP_CHECK"
```

Comment:
why `clean` here?
breaks incremental build

Reply:
Yea, not needed.